rust/libnewsboat/src/matcher.rs

```diff
@@ -21,16 +25,960 @@ impl Matcher {
     ///
     /// If `input` can't be parsed, returns a user-readable error.
     pub fn parse(input: &str) -> Result<Matcher, String> {
-        unimplemented!()
+        let expr = filterparser::parse(input).map_err(|e| match e {
+            filterparser::Error::TrailingCharacters(tail) => {
+                format!("Parse error: trailing characters: {}", tail)
+            }
+            filterparser::Error::AtPos(pos) => format!("Parse error at position {}", pos),
```

Comment:
Do you mean "which sub-parser failed"? I do plan to make the errors more useful (https://github.com/newsboat/newsboat/issues/631#issuecomment-616196512), but that's a chunk of work in itself. It will be a separate PR. The errors will be propagated from here to `parser`, up to `Matcher` object which will finally return it from its `parse()` method — the same as current C++ implementation does.

Reply:
Sounds good!